### DISCUSSION TITLE
GPU version of the program cannot load the model

### DISCUSSION THREAD

**Discussion by zdpaa** [vietanhdev/anylabeling#77] [Q&A]
I have downloaded the dependencies and the corresponding CUDA for the GPU version, but after running it, the automatic annotation model does not load
![image](https://user-images.githubusercontent.com/94962587/236672123-87a72b56-b3b0-4cac-bfbe-77d9a81e437a.png)

**Comment by vietanhdev** [comment]
Please specify:
- Your OS
- Your PC hardware (CPU/GPU)
- The steps you used to install AnyLabeling
- The error message

Please check the instructions to use GPU here: https://anylabeling.com/docs/gpu.

**Reply by zdpaa** [reply]
Still not working, you see the results here
![image](https://user-images.githubusercontent.com/94962587/236673497-9de9b428-63e1-411f-ad66-1d6a3213ab7b.png)

**Reply by vietanhdev** [reply]
Do you see any error massage in the Terminal?

**Reply by vietanhdev** [reply]
Could you please check if CPU mode has the same issue on your machine? Thank you!

**Reply by zdpaa** [reply]
> Do you see any error massage in the Terminal?

no

**Reply by zdpaa** [reply]
> Could you please check if CPU mode has the same issue on your machine? Thank you!

![image](https://user-images.githubusercontent.com/94962587/236675391-3a89d5be-b499-4751-a5f5-cf3963706bff.png)
this is with cpu，it can work

**Reply by vietanhdev** [reply]
@zdpaa In the release note of ONNX Runtime 1.14.1, they support CUDA 12. Which version are you using on your machine?

**Reply by zdpaa** [reply]
> @zdpaa In the release note of ONNX Runtime 1.14.1, they support CUDA 12. Which version are you using on your machine?

cuda is 11.6 and cudnn is 8.5

**Reply by zdpaa** [reply]
> @zdpaa In the release note of ONNX Runtime 1.14.1, they support CUDA 12. Which version are you using on your machine?
ONNX Runtime 1.14.1 only support CUDA 12.？

**Reply by zdpaa** [reply]
> ntime 1.14.1 only support CUDA 12.？

![image](https://user-images.githubusercontent.com/94962587/236682502-83449d8a-591d-4024-b960-baebb808dcf1.png)

**Reply by vietanhdev** [reply]
@zdpaa  Could you help me to try ONNXRuntime 1.14.0?
```
pip install onnxruntime-gpu==1.14.0
```

**Reply by zdpaa** [reply]
> @zdpaa Could you help me to try ONNXRuntime 1.14.0?
> 
> ```
> pip install onnxruntime-gpu==1.14.0
> ```

**Comment by zdpaa** [comment]
I tried 1.14.1 and 1.14.0 they are fine, but maybe my computer graphics card configuration is not enough, the model can be loaded, but after loading to mark will keep spinning and then automatically exit
![image](https://user-images.githubusercontent.com/94962587/236683739-6b63593a-8206-4aeb-9e18-cc3b17cefcac.png)
My gpu is 3060, 6g video memory
The cpu version can instead be used normally, although sometimes there will be an automatic exit

**Reply by vietanhdev** [reply]
@zdpaa My SAM B quantized model takes 7.2GB of GPU memory when running on RTX 2070. Your GPU memory may not be enough.
However, the issue may be because the model download was not successful. I will release v0.2.23 to handle model downloading better. In the meantime, you may want to download the models from [here](https://github.com/vietanhdev/anylabeling-assets/releases/tag/v0.4.0), extract and load them into your program using instructions from [this page](https://anylabeling.com/docs/custom-models).
Thank you for your feedback!
